src/logging.c

```diff
@@ -104,47 +173,149 @@ void rcutils_logging_console_output_handler(
       return;
   }
 
-  char buffer[1024];
-  char * message_buffer = buffer;
+  // Declare variables that will be needed for cleanup ahead of time.
```

Comment:
if this makes the code less readable we can revert https://github.com/ros2/rcutils/pull/53/commits/18777e23224232338ece56d0d1c4de590b469803 and not use `cleanup` on message format failure.